internal/version/version.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2020, Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package version
+
+// Version variable will be replaced at link time after `make` has been run.
```

Comment:
Why are these two separate strings and not just one?

Reply:
Well, because right now you're constructing a version that doesn't actually match the package version. e.g. our Debian package right now is `1.0.5~debian10` but you're going to make `1.0.5-debian10`.

Reply:
The syntax of useragent is defined here: b170441793#comment15. We are not going for `1.0.5~debian10` as the package version..

Reply:
We should not be inventing a new version number format just for this use case. We should either be reporting the code version (`1.0.5`) or the package version (`1.0.5~debian10` or `1.0.5-1`). I think I would prefer the package version (because it contains other info not reported elsewhere - e.g. the package revision), but code vesion would also be defensible. If you want the distro the package was built on separately, you should report that as a separate field, not as part of a bespoke version number format.

Reply:
I don't think the intention was not to invent a new version number format.  Beloow format has already been decided after rounds of serious discussion i suppose.

```
Google-Cloud-Ops-Agent-Collector/<OPS_AGENT_VERSION>-<BUILD_DISTRO> (OPS_AGENT_COMMENT),gzip(gfe)
```
The useragent will later be used for internal analysis. For example, by looking at the above format, i suppose there wlll analysis of obtaining the build_distro. Thus it makes sense to use separator in this case it's `hyphen` to distinguish it from the version

Reply:
`1.0.5-stretch` is a new version number format. That version number is not used anywhere else.

Reply:
If `-` makes it look like a new version number format, shall we re-discussion this part `<OPS_AGENT_VERSION>-<BUILD_DISTRO>`.  We would be very likely to have an analysis that's to only grab `1.0.5` and see how many VMs uses that version.. So it makes sense to have those two in separate fields

Reply:
Google-Cloud-Ops-Agent-Logging/1.0.5 (BuildDistro=9;Platform=Linux;ShortName=Debian;ShortVersion=9 is acceptable accordng to ling